mergify_engine/web/simulator.py

```diff
@@ -126,7 +126,13 @@ async def simulator(request: requests.Request) -> responses.JSONResponse:
     if token:
         token = token[6:]  # Drop 'token '
 
-    data = SimulatorSchema(await request.json())
+    data = await request.json()
+
+    if data.get("mergify.yml"):
+        data["mergify.yml"] = rules.YamlSchema(data["mergify.yml"])
```

Comment:
You should update the `SimulatorSchema` instead of modifying data before the validation.